src/test/java/com/opipo/codechallenge/service/impl/TransactionStatusServiceImplTest.java

```diff
@@ -0,0 +1,85 @@
+package com.opipo.codechallenge.service.impl;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.opipo.codechallenge.repository.TransactionRepository;
+import com.opipo.codechallenge.repository.model.TransactionEntity;
+import com.opipo.codechallenge.service.TransactionStatusChannelService;
+import com.opipo.codechallenge.service.TransactionStatusDateService;
+import com.opipo.web.api.model.TransactionStatusRequest;
+import com.opipo.web.api.model.TransactionStatusRequest.ChannelEnum;
+import com.opipo.web.api.model.TransactionStatusResponse;
+import com.opipo.web.api.model.TransactionStatusResponse.StatusEnum;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("TransactionStatusServiceImpl test")
+public class TransactionStatusServiceImplTest {
+
+    @InjectMocks
+    private TransactionStatusServiceImpl transactionStatusServiceImpl;
+
+    @Mock
+    private TransactionRepository transactionRepository;
+
+    @Mock
+    private TransactionStatusDateService transactionStatusDateService;
+
+    @Mock
+    private TransactionStatusChannelService transactionStatusChannelService;
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @DisplayName("Given transactionStatusRequest with inexistent referente then return status invalid")
+    void givenTransactionStatusRequestWithInexistentReferenceThenReturnInvalid() {
+        String reference = "myId";
+        TransactionStatusRequest transactionStatusRequest = new TransactionStatusRequest();
+        transactionStatusRequest.setReference(reference);
+        Mockito.when(transactionRepository.findById(reference)).thenReturn(Optional.empty());
+        TransactionStatusResponse actual = transactionStatusServiceImpl.execute(transactionStatusRequest);
+        assertNotNull(actual);
+        assertEquals(reference, actual.getReference());
+        assertEquals(StatusEnum.INVALID, actual.getStatus());
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)